GiEnJul/ClientApp/src/components/registerGiver/SignUp.tsx

```diff
@@ -1,179 +1,188 @@
-    const [email, setEmail] = useState<string | undefined>(undefined);
-    const [phoneNumber, setPhoneNumber] = useState<string | undefined>(undefined);
-    const [maxRecievers, setMaxRecievers] = useState<number | undefined>(undefined);
-    const [familyType, setFamilyType] = useState<string | undefined>(undefined);
+  const handlefullnameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setInputsState((prev) => {
+      return { ...prev, fullname: event.target.value };
+    });
+  };
 
-    const [locationOptions, setLocationOptions] = useState<string[]>([]);
+  const handleEmailChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setInputsState((prev) => {
+      return { ...prev, email: event.target.value };
+    });
+  };
 
-    useEffect(() => {
-        getLocations().then((locationArray) => setLocationOptions(locationArray))
-     }, []);
+  const handleTlfChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setInputsState((prev) => {
+      return { ...prev, phoneNumber: event.target.value };
+    });
+  };
 
-    // go back to previous step
-    const prevStep = () => {
-        setStep(step - 1);
-    };
-    // proceed to the next step
-    const nextStep = () => {
-        setStep(step + 1);
-    };
-    const handlefullnameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-        setFullname(event.target.value)
-    };
-    const handleEmailChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-        setEmail(event.target.value)
-    };
-    const handleTlfChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-        setPhoneNumber(event.target.value)
-    };
-    const handleLocationChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-        setLocation(event.target.value)
-    }
-    const handleFamilyChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
-        var value = event.target.value;
-        setFamilyType(value)
-        if (value === 'Liten familie') {
-            setMaxRecievers(2)
-        }
-        if (value === 'Vanlig familie') {
-            setMaxRecievers(5)
-        }
-        if (value === 'Stor familie') {
-            setMaxRecievers(100)
-        }
-    }
-    const handleConfirm = (submitOK: boolean) => {
-        setConfirmationOK(submitOK)
-    }
+  const handleLocationChange = (
+    event: React.ChangeEvent<HTMLSelectElement>
+  ) => {
+    setInputsState((prev) => {
+      return { ...prev, location: event.target.value };
+    });
+  };
 
-    const values = { location, fullname, email, phoneNumber, maxRecievers, familyType, confirmationOK  };
-    const submit = { location, fullname, email, phoneNumber, maxRecievers, confirmationOK }
-    const familiyOptions = ['Liten familie', 'Vanlig familie', 'Stor familie'];
+  const handleFamilyChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setInputsState((prev) => {
+      return { ...prev, maxReceivers: event.target.value };
+    });
+  };
```

Comment:
All handleChange functions can be rewritten in a much simpler way, this is just making things less repetitive.

const handleOnChange = (target, value) => {
   setState((prev) => ({ ...prev, [target]: value, })) 
}

Notice no return function needed.